### DISCUSSION TITLE
Implementing ITaskHandler

### DISCUSSION THREAD

**Discussion by CCob** [dahall/TaskScheduler#877] [Q&A] [answered]
Hi There,

Any idea what I could be missing when registering a C# Library with an exposed COM object that implements ITaskHandler?  Currently I get error 0x80040154 (REGDB_E_CLASSNOTREG) when attempting to start a task within task scheduler.  

* Task created with IComHandlerAction and CLSID set to COM object exposed via .NET COM Type
* Strong signed assembly
* Assembly is COM Visible
* Assembly set to ActivationOption.Server
* CLSID and AppId registered within the registry 

I can create the same COM object via PowerShell no problem (but it seems to be inproc which could be the issue).  But task scheduler refuses to run the task.

Any ideas would be great.
Thanks.

**Comment by dahall** [comment, marked as answer]
Did you already look here: https://github.com/dahall/ITaskHandlerTemplate ?

It is a fully implemented C# project to create a handler. You may get some hints for your project config and class attributes by looking at that project and its files.

**Reply by CCob** [reply]
Yes, my entire project is based on the template, but I have created it from scratch.  The end goal is to register the COM object within CKCU\Software\Classes sub key as opposed to HKCR.  But I can't get it working via HKCR either at the moment.  Is there a checklist of things I should be looking for.  Also, is COM+ a requirement for via COM based tasks?

Thanks

**Reply by CCob** [reply]
OK, It worked via RegSvcs.exe but not RegAsm.exe.  Do you happen to know what difference there is when using regsvcs as I need to implemented the registration programmatically.

Thanks.

**Comment by dahall** [comment]
RegAsm deals with registering managed classes as COM visible. RegSvcs does a lot more (see [docs](https://docs.microsoft.com/en-us/dotnet/framework/tools/regsvcs-exe-net-services-installation-tool)). COM+ is not necessarily a requirement for implementing the in-proc server, but it is the only way that I have found to get a .NET assembly to work.
